cheatsheets/Nodejs_security_cheat_sheet.md

```diff
@@ -0,0 +1,534 @@
+
+- **Strict-Transport-Security**: [HTTP Strict Transport Security (HSTS)](https://cheatsheetseries.owasp.org/cheatsheets/HTTP_Strict_Transport_Security_Cheat_Sheet.html) dictates browsers that the application can only be accessed via HTTPS connections. This header takes two parameters:  `max-age` to determine how long this configuration will be valid and `includeSubDomains` to state if subdomains are to be treated in the same way. In order to use it in your application, add the following codes:
+
+```JavaScript
+app.use(helmet.hsts()); // default configuration
+app.use(helmet.hsts("<max-age>", "<includeSubdomains>")); // custom configuration
+```
+
+- **X-Frame-Options:** determines if a page can be loaded via a \<frame> or an \<iframe> element. Allowing the page to be framed may result in clickjacking attacks which aims to manipulate users into clicking on a different element instead of the one they intend to. This header has 3 directives: DENY to never allow framing, SAMEORIGIN to only allow framing within the same origin and ALLOW-FROM to only allow framing from specified URIs. These behaviors can be achieved with helmet module as follows:
+
+```JavaScript
+app.use(hemlet.xframe()); // default behavior (DENY)
+helmet.xframe('sameorigin'); // SAMEORIGIN
+helmet.xframe('allow-from', 'http://alloweduri.com'); //ALLOW-FROM uri
+```
+
+- **X-XSS-Protection:** As described in [XSS Prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#bonus-rule-4-use-the-x-xss-protection-response-header), this header enables browsers to stop loading pages when browsers detect reflected cross-site scripting attacks. In order to implement this header in your application, you can use the following code:
+
+```JavaScript
+var xssFilter = require('x-xss-protection');
+app.use(xssFilter());
+```
+
+- **X-Content-Type-Options:** Even if the server sets a valid Content-Type header in the response, browsers may try to sniff the MIME type of the requested resource. This header is a way to stop this behavior and tell the browser not to change MIME types specified in Content-Type header. It can be configured in the following way:
+
+```JavaScript
+app.use(helmet.noSniff());
+```
+
+- **Content-Security-Policy:** Content Security Policy is developed to reduce the risk of attacks like XSS and Clickjacking. Basically, it allows content from a whitelist you decide. Other content from different sources is not accepted if Content-Security-Policy headers are set correctly. It has several directives each of which prohibits loading specific type of a content. These are `connect-src`, `font-src`, `frame-src`, `img-src`, `media-src`, `object-src`, `script-src`, `style-src` and `default-src`. These can be assigned to self, none, unsafe-inline or unsafe-eval. You can implement these settings in your application as follows:
```

Comment:
You can link to the CSP CS instead of describing it yourself 😊 This is a CS, shouldn't tackle every subject in details.